### ISSUE TITLE
Support `MultiArgumentInputs` for Resource Methods

### ISSUE THREAD

**Issue by flostadler** [pulumi/pulumi#16563]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Right now resource methods do not support `MultiArgumentInputs`, the flag is ignored for codegen & docsgen. Instead an `args` param is created for TS, CS and Go.
E.g.: https://github.com/pulumi/pulumi/blob/39c5647849857bd862814dce76136406f3967ecb/pkg/codegen/docs/gen_method.go#L121

Supporting `MultiArgumentInputs` would allow schematizing existing Overlays like S3 `Bucket:onObjectCreated` which helps documenting it and could be used in the future to turn them into MLCs.

**Comment by justinvp** [comment]
> schematizing existing Overlays like S3 `Bucket:onObjectCreated`

Just a note that there are some other challenges to schematizing this. We don't currently have a good way of representing `BucketEventHandler` or `pulumi.ComponentResourceOptions` as arguments.

**Comment by flostadler** [comment]
> Just a note that there are some other challenges to schematizing this. We don't currently have a good way of representing `BucketEventHandler` or `pulumi.ComponentResourceOptions` as arguments.

Yes that's true, but we could model those exceptions as `any` types and add more details in the description/examples. At least that would allow us to add them to the docs of their corresponding resources.
